File: WindowsGrep/WindowsGrep.Engine/Data/SearchMetrics.cs
namespace WindowsGrep.Engine;

public class SearchMetrics
{
    #region Properties..
    public int TotalFilesMatchedCount { get; set; }

    public int DeleteSuccessCount { get; set; }

    public int ReplacedSuccessCount { get; set; }

    public List<FileItem> FailedReadFiles { get; set; } = new List<FileItem>();

    public List<FileItem> FailedWriteFiles { get; set; } = new List<FileItem>();
    #endregion Properties..
}

File: WindowsGrep_Rust/src/models/search_metrics.rs
use crate::models::file_item::FileItem;

#[derive(Debug, Default)]
pub struct SearchMetrics {
    pub total_files_matched_count: i32,
    pub delete_success_count: i32,
    pub replaced_success_count: i32,
    pub failed_read_files: Vec<FileItem>,
    pub failed_write_files: Vec<FileItem>,
}
